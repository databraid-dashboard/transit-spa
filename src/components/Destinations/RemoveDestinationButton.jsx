import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { removeDestination } from '../../actions';

class RemoveDestinationButton extends React.Component {

  buttonWasClicked = (e) => {
    e.preventDefault();
    this.props.removeDestination(this.props.id);
  }

  render() {
    return (
      <Button
        icon="window close"
        size="mini"
        float="right"
        onClick={this.buttonWasClicked}
      />
    );
  }
}

RemoveDestinationButton.propTypes = {
  removeDestination: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators({
  removeDestination,
}, dispatch);

export default connect(
  null,
  mapDispatchToProps,
)(RemoveDestinationButton);
