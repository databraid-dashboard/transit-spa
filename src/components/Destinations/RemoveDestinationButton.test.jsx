import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RemoveDestinationButton, mapDispatchToProps } from './RemoveDestinationButton';

describe('RemoveDestinationButton', () => {
  it('should render a remove button', () => {
    const id = 1;
    const component = shallow(<RemoveDestinationButton id={id} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('removeDestination');
  });
});
