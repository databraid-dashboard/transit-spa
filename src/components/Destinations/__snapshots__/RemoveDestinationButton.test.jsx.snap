// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveDestinationButton should render a remove button 1`] = `
<Button
  as="button"
  circular={true}
  color="red"
  floated="right"
  icon="delete"
  onClick={[Function]}
  size="small"
/>
`;
